components/Search.tsx

```diff
@@ -1,19 +1,26 @@
 "use client";
-import React from "react";
+import { useState, useEffect } from "react";
+import { debounce } from "lodash";
 import SearchBar from "./SearchBar";
 import SearchResult from "./SearchResult";
 
 export default function Search() {
-  const [searchValue, setSearchValue] = React.useState<string>("");
+  const [clientSearchValue, setClientSearchValue] = useState<string>("");
+  const [serverSearchValue, setServerSearchValue] = useState<string>("");
+
+  useEffect(
+    () => debounce(() => setServerSearchValue(clientSearchValue), 700),
```

Comment:
this isn't working right, typing more isn't resetting the delay. You probably need a useCallback for the function you're sending, I'd read the react docs about it